tests/utils/streams/mixins.py

```diff
@@ -0,0 +1,159 @@
+import uuid
+from abc import ABC, abstractmethod
+from concurrent.futures import wait
+from contextlib import closing
+from typing import ContextManager, Mapping, Sequence
+
+import pytest
+
+from snuba.utils.streams.consumer import Consumer, ConsumerError, EndOfPartition
+from snuba.utils.streams.producer import Producer
+from snuba.utils.streams.types import Message, Partition, Topic
+
+
+class StreamsTestMixin(ABC):
+    @abstractmethod
+    def get_topic(self) -> ContextManager[Topic]:
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_consumer(self, group: str) -> Consumer[int]:
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_producer(self) -> Producer[int]:
+        raise NotImplementedError
+
+    def test_consumer(self) -> None:
```

Comment:
There are some things in this method that don't pass type checks (mostly methods on `KafkaConsumer` that are not part of the base `Consumer`), I will follow up with those improvements later.

Reply:
Also, all of these offsets are still coupled with any consumer backend using sequential offsets (namely, the Kafka and dummy consumers), these will have to change with the file consumer to use the offsets returned in the future returned by `produce`.